src/main/java/com/flansmod/common/driveables/EntityDriveable.java

```diff
@@ -645,57 +635,43 @@ private void shootEach(DriveableType type, DriveablePosition shootPoint, int cur
 			}
 			
 			// For each 
+
 			ItemShootable shootableItem = (ItemShootable)shootableStack.getItem();
 			ShootableType shootableType = shootableItem.type;
 			
-			float spread = 0.005f * gunType.bulletSpread * shootableType.bulletSpread;
-			
-			lookVector.x += (float)world.rand.nextGaussian() * spread;
-			lookVector.y += (float)world.rand.nextGaussian() * spread;
-			lookVector.z += (float)world.rand.nextGaussian() * spread;
+			if (!gunType.isAmmo(shootableType))
+				return;
 			
-			lookVector.scale(500.0f);
+			FireableGun fireableGun = new FireableGun(gunType, gunType.damage, gunType.bulletSpread, gunType.bulletSpeed);
+			//TODO unchecked cast, grenades will cause a crash (currently no vehicle with this feature exists)
```

Comment:
Can't we just catch it, error and return?

Reply:
I have inserted a casting check for good coding practices, but i plan on reworking grenades anyways which would completely resolve all problems related to them